src/main/ui/app/component/SearchBox.jsx

```diff
@@ -21,43 +22,57 @@ class SearchBox extends Component {
 
     setQuery(query) {
         this.setState({
-                          query: query
-                      });
+            query: query
+        });
     }
 
     doSearch() {
-        if (this.state.type || this.state.query) {
+        if (this.state.query) {
+            let query = this.state.query.toLowerCase();
+            const options = {keywords: ['uid', 'type', 'tag', 'query', 'moduleid', 'status', 'name'], offsets: false};
+            //Basic Regex Matching to fix query inconsistencies
+            //Removing all extra spaces i.e. two or more
+            query = query.replace(/\s+/g,' ').trim();
+            //Removing all spaces to the left of search keys
+            query = query.replace(new RegExp("\\s+:","g"),":");
+            //Removing all spaces to the right of search keys
+            query = query.replace(new RegExp(":\\s+","g"),":");
             let url = "/search?";
-            if (this.state.type) {
-                url += "type=" + this.state.type;
+            // Check if query is an advanced query. We want to use the Parser only if the query is advanced
+            if (query.includes(":") || query.includes("-")) {
+                let searchQueryObj = searchQuery.parse(query, options);
+                Object.keys(searchQueryObj).forEach(function (key) {
+                    console.log(key);
+                    if (searchQueryObj[key]){
+                        if (key === "text" || key === "query") {
+                            url += "&q=" + searchQueryObj[key];
+                        }
+                        else if (key === "type"){
+                            url += "&type=" + searchQueryObj[key].toUpperCase();
+                        }
+                        else if (key === "exclude"){
+                            if (searchQueryObj[key].text){
+                                url += "&exclude=" + searchQueryObj[key].text;
+                            }
+                        }
+                        else if (key === "status"){
```

Comment:
you're handling "type" and "status" in the same way, can you combine these branches?